### PR TITLE
chore(layer): demo issue with context from different modules

### PR DESCRIPTION
**Pull request by HendrikThePendric**
This is not a real PR, it's just a minimalistic demo of the issue @martinkrulltott found....

We should look into a way of creating the context differently....

### CODE REVIEW

File: packages/widgets/src/LayerContext/LayerContextTwo.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * @static
+ * @prop {function} children
+ * @prop {number} zIndex
+ */
+LayerTwo.propTypes = {
+    children: propTypes.func.isRequired,
+    zIndex: propTypes.number,
+}
+
+export { LayerTwo, useLayer }
```
Comment (HendrikThePendric) [inline packages/widgets/src/LayerContext/LayerContextTwo.js]: This is just a duplicate of the `Layer` from `LayerContext.js` but with a different name.

File: packages/widgets/src/LayerContext/LayerContext.stories.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        {zIndexComputed => (
+            <>
+                <h1>{zIndexComputed}</h1>
+                <LayerTwo zIndex={FIXED_INDEX}>
+                    {anotherZIndexComputed => <h1>{anotherZIndexComputed}</h1>}
+                </LayerTwo>
+            </>
+        )}
+    </Layer>
+)
```
Comment (HendrikThePendric) [inline packages/widgets/src/LayerContext/LayerContext.stories.js]: It goes wrong here, because each layer is referring to a different base context.

**Comment by HendrikThePendric** [comment]
In https://github.com/dhis2/ui/pull/16/commits/1e5c9036bb85f49ed633f4cfa4599c9d87e734c5 I've found a "fix" for this problem. The only way I could think of was to attach the context to the window to avoid duplicates from being created. It's definitely not the most elegant solution available, but I couldn't think of another way to solve it. Any other suggestions would be most appreciated....

Personally I can't think of any other way to solve this, apart from not allowing several instances of ui-core from being used in one app.... Which, to be fair, probably is something worth discussing further anyway....

**Comment by ismay** [comment]
Are we looking for an intermediate solution here, while we figure out a long term solution to the whole zindex issue? Just for some context on the solution we're looking for, because as an intermediary solution something dirty seems less problematic.

**Comment by HendrikThePendric** [comment]
This pull request was just there to illustrate https://github.com/dhis2/ui/issues/15 which is now closed, so we can close this PR too.

For the record: in the end we decided not to fix this in our library, but to let the app prevent this problem from occurring, see https://github.com/dhis2/ui/pull/17.

**Pull request closed without merging**